docs/guides/lint-asyncapi.md

```diff
@@ -0,0 +1,133 @@
+specification.
+
+With this configuration file, and your AsyncAPI description file, run the linting command:
+
+```
+redocly lint asyncapi.yaml
+```
+
+The output describes any structural problems with the document, or reports that it is valid.
+
+
+## AsyncAPI rules
+
+To expand the linting checks for an AsyncAPI description, start by enabling
+some of the built-in rules. The currently-supported rules are:
+
+* `info-contact`: the `Info` section must contain a valid `Contact` field.
+* `operation-operationId`: every operation must have a valid `operationId`.
+* `channels-kebab-case`: channel names should be `kebab-case` (lowercase with hyphens)
+* `no-channel-trailing-slash`: channel names must not have trailing slashes in their names
+* `tag-description`: all tags require a description
+* `tags-alphabetical`: tags should be listed in the AsyncAPI file in alphabetical order
+
+We expect the list to expand over time, so keep checking back - and let us know
+if you have any requests by [opening an issue on the GitHub
+repo](https://github.com/Redocly/redocly-cli/issues).
+
+To use a rule in your own linting setup, add the rule name to the `rules`
+configuration section, and declare the severity level (either `error`, `warn`
+or `off`). Here's an example of a rules block where a missing contact section
+causes a warning, and a tag without a description triggers an error:
+
+```yaml
+rules:
+  info-contact: warn
+  tag-description: error
+```
+
+Pick and mix the available rules until you have the setup that fits your situation.
+
+## Configurable rule example
+
+Redocly CLI also offers [configurable rules](../rules/configurable-rules.md)
+that allow you to set assertions about the API description being linted, and
+this functionality works just as well for AsyncAPI as for OpenAPI. The
+following example shows a configurable rule that emits a warning if the `title`
+field is not present in the `info` block:
+
+```yaml
+rules:
+  rule/info-title:
+    subject:
+      type: Info
+      property: title
+    assertions:
+      defined: true
+    severity: warn
+```
+
+With the extensive configurable rules options available, there are many
+opportunities to make sure that your AsyncAPI spec conforms with expectations
+(we'd also love to see what you're building, it helps us know how things are
+going!).
+
+:::info Custom plugins
+
+For those users with advanced requirements and JavaScript skills, the [custom
+plugins](../custom-plugins/index.md) feature offers some extension points if you need
+them.
+
+:::
+
+## Supported protocols
+
+AsyncAPI supports an ever-expanding list of protocols, here's the list of what's currently supported:
+
+* `http`
+* `ws`
+* `kafka`
+* `anypointmq`
+* `amqp`
+* `amqp1`
+* `mqtt`
+* `mqtt5`
+* `nats`
+* `jms`
+* `sns`
+* `solace`
+* `sqs`
+* `stomp`
+* `redis`
+* `mercure`
+
+If you're using other protocols, you can still use Redocly CLI to lint an
+AsyncAPI description, but the details of those protocol bindings aren't
```

Comment:
Do we also want users to submit requests for protocols they'd like to see coverage for? Would it be possible for users (or maybe vendors) to submit PRs adding support for new protocols?

Reply:
I didn't call for it because it's kind of a PITA to create right now, and really it would be better for vendors to produce custom plugins, but until we fix [#1223](https://github.com/Redocly/redocly-cli/issues/1223) that's less useful. It's the three most recently added protocols that are missing I believe - and I'd really love to be encouraging more community contribution but we need to release something first...